File: BackEnd/GPON_Equipment_MicroService/src/main/java/com/example/gpon_equipment_microservice/Service/ONT/ONTService.java
package com.example.gpon_equipment_microservice.Service.ONT;

import com.example.gpon_equipment_microservice.Entity.ONT.ONT;
import com.example.gpon_equipment_microservice.Entity.Status;
import com.example.gpon_equipment_microservice.Repository.ONTRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class ONTService implements IONTService {
    private final ONTRepository ontRepository;

    @Override
    public ONT save(ONT ont) {
        return ontRepository.save(ont);
    }

    @Override
    public List<ONT> findAll() {
        return ontRepository.findAll();
    }

    @Override
    public Optional<ONT> findById(Long id) {
        return ontRepository.findById(id);
    }

    @Override
    public List<ONT> findByStatus(Status status) {
        return ontRepository.findByStatus(status);
    }

    @Override
    public ONT findByAddressId(Long addressId) {
        return ontRepository.findByAddressId(addressId);
    }

    @Override
    public ONT findByClientId(Long clientId) {
        return ontRepository.findByClientId(clientId);
    }

    @Override
    public List<ONT> findByConnectedToBoxId(Long boxId) {
        return ontRepository.findByConnectedToBoxId(boxId);
    }

    @Override
    public List<ONT> findByBoardId(Long boardId) {
        return ontRepository.findByBoardId(boardId);
    }

    @Override
    public List<ONT> findByOltId(Long oltId) {
        return ontRepository.findByOltId(oltId);
    }
    
    @Override
    public void deleteById(Long id) {
        ontRepository.deleteById(id);
    }
}

File: FrontEnd/src/Services/Equipment/ONT/ont-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environment/environment';
import { ONT } from '../../../Core/Models/Equipment/ONT';
import { Status } from '../../../Core/Models/Equipment/Status';

@Injectable({
  providedIn: 'root'
})
export class ONTService {
  private apiUrl = `${environment.BASE_URL}/equipment/api/ont`;

  constructor(private http: HttpClient) { }

  // Get all ONTs
  findAll(): Observable<ONT[]> {
    return this.http.get<ONT[]>(`${this.apiUrl}/all`);
  }

  // Get ONT by ID
  findById(id: number): Observable<ONT> {
    return this.http.get<ONT>(`${this.apiUrl}/${id}`);
  }

  // Get ONTs by status
  findByStatus(status: Status): Observable<ONT[]> {
    return this.http.get<ONT[]>(`${this.apiUrl}/status/${status}`);
  }

  // Get ONT by address ID
  findByAddressId(addressId: number): Observable<ONT> {
    return this.http.get<ONT>(`${this.apiUrl}/address/${addressId}`);
  }

  // Get ONT by client ID
  findByClientId(clientId: number): Observable<ONT> {
    return this.http.get<ONT>(`${this.apiUrl}/client/${clientId}`);
  }

  // Get ONTs by box ID
  findByBoxId(boxId: number): Observable<ONT[]> {
    return this.http.get<ONT[]>(`${this.apiUrl}/box/${boxId}`);
  }

  // Get ONTs by board ID
  findByBoardId(boardId: number): Observable<ONT[]> {
    return this.http.get<ONT[]>(`${this.apiUrl}/board/${boardId}`);
  }

  // Get ONTs by OLT ID
  findByOltId(oltId: number): Observable<ONT[]> {
    return this.http.get<ONT[]>(`${this.apiUrl}/olt/${oltId}`);
  }

  // Save new ONT
  save(ont: ONT): Observable<ONT> {
    return this.http.post<ONT>(`${this.apiUrl}/save`, ont);
  }

  // Update existing ONT
  update(id: number, ont: ONT): Observable<ONT> {
    return this.http.put<ONT>(`${this.apiUrl}/update/${id}`, ont);
  }

  // Delete ONT
  delete(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/delete/${id}`);
  }
}
